common_clang.cpp

```diff
@@ -248,7 +250,9 @@ Compile(const char *pszProgramSource, const char **pInputHeaders,
 
       // Calling ResetAllOptionOccurrences as WA for issue from here:
       // https://reviews.llvm.org/D66324?id=219733#1680231
+      Diags->setSeverity(clang::diag::warn_profile_data_misexpect, clang::diag::Severity::Warning, clang::SourceLocation());
       llvm::cl::ResetAllOptionOccurrences();
```

Comment:
My question is not relevant to this patch:
Does the above error message means we don't need ResetAllOptionOccurrences for 120 and master branch any more? Since the issue is already fixed in master branch of llvm.

Reply:
I dont know which LLVM version the master is based on.
llvm 9 & llvm12 : no need this patch. 
llvm 10 and 11 need. 
It is because the relevant patch (adding warn_profile_data_misexpect not properly) is included in llvm 10 and llvm 11, and then reverted in llvm 12.

Reply:
sorry I misunderstood. I thought ResetAllOptionOccurrences was not added by you.

Reply:
git log shows ResetAllOptionOccurrences was added by @lwesiers, I think we may remove it in 120 and master branch since the guilty patch is reverted in llvm 120 and llvm main branch.